### PR TITLE
Fit integer tune

### PR DESCRIPTION
**Pull request by swhite2401**
This PR adds to possibility to fit the integer part of the tune either by set `fit_integer=True` as argument or providing a non-zero integer part for one of the target values.
Follow-up from #456

### CODE REVIEW

**Comment by simoneliuzzo** [comment]
Dear @swhite2401 , thank you for this addition. The function get_tune may be should have the same optional input? 
for example, to check that the tune correction is correctly done I would do:
```
initial_tune = get_tune(ring, ..., use_integer = True)
ring_tune = at.matching.fit_tune(ring, ...,fit_integer=True)
final_tune = get_tune(ring_tune, ..., use_integer = True)
```

**Comment by simoneliuzzo** [comment]
Dear @swhite2401 

the help of at.matching.globalfit.fit_tune gives examples that are may be not up to date.
The help of get_tune could be complemented with the get_integer optional flag

**Comment by simoneliuzzo** [comment]
works for me

**Review by simoneliuzzo — APPROVED** [review]

File: pyat/at/matching/globalfit.py [hunk 1]
```diff
@@ -4,91 +4,105 @@
 """
 from typing import Optional
 import numpy
+from warnings import warn
 from ..lattice import Lattice, Refpts, set_value_refpts
+from ..lattice import AtWarning, AtError
```
Comment (lfarv) [inline pyat/at/matching/globalfit.py:9]: I have "Unused import statement" warnings. Apparently warn is not used… Are these lines left after tests?

File: pyat/at/matching/globalfit.py [hunk 2]
```diff
@@ -4,91 +4,105 @@
...
         refpts1:    Selection of the 1st family
         refpts2:    Selection of the 2nd family
-        newval:     New tunes
+        newval:     New tunes, in case an non-zero integer part
+                    is provided, fit_integer is set to True
         tol:        Tolerance for the matching; Default: 1.0e-12
         dp:         Momentum deviation. Default: 0
         niter:      Maximum number of iterations. Default 3
-    
+        fit_integer: bool (default=False), use integer tune
```
Comment (lfarv) [inline pyat/at/matching/globalfit.py:90]: `fit_integer` is not in the argument list. It should be in a "Keyword Arguments:" section.

**Pull request merged**